src/megaapi_impl.cpp

```diff
@@ -8374,34 +8354,30 @@ int MegaApiImpl::syncPathState(string* path)
     // once we do manage to lock, return to normal operation.
     syncPathStateLockTimeout = false;
 
+    LocalPath localpath = LocalPath::fromLocalname(*path);
```

Comment:
this could be avoided if !client->syncs.size():
[untested]:

```suggestion
    if (!client->syncs.size())
    {
        return state;
     }
    LocalPath localpath = LocalPath::fromLocalname(*path);
```

Reply:
good point, this is a case where we want to be very quick, responding to the file manager, especially when there are no syncs - added, thanks